src/hooks/handlers/handleOrderSubmit.ts

```diff
@@ -0,0 +1,70 @@
+import { toast } from 'react-toastify';
+import { ROUTES } from '@/constants/routes';
+import { ERROR_MESSAGES } from '@/constants/validation';
+import { hasAxiosErrorStatus } from '@/utils/error';
+import type { OrderFormValues } from '@/types/order';
+import type { ProductSummary } from '@/hooks/useProductSummary';
+import type { NavigateFunction } from 'react-router-dom';
+
+interface HandleOrderSubmitParams {
+  data: OrderFormValues;
+  product: ProductSummary;
+  selectedCardId: number;
+  authToken: string;
+  totalQuantity: number;
+  navigate: NavigateFunction;
+  mutateAsync: (payload: {
+    formData: OrderFormValues;
+    productId: number;
+    messageCardId: string;
+    authToken: string;
+  }) => Promise<unknown>;
+}
+
+export const handleOrderSubmit = async ({
+  data,
+  product,
+  selectedCardId,
+  authToken,
+  totalQuantity,
+  navigate,
+  mutateAsync,
+}: HandleOrderSubmitParams) => {
+  if (data.receivers.length === 0) {
+    toast.error(ERROR_MESSAGES.EMPTY_RECEIVERS);
+    return;
+  }
+
+  try {
+    await mutateAsync({
+      formData: data,
+      productId: product.id,
+      messageCardId: String(selectedCardId),
+      authToken,
+    });
+
+    alert(
+      `주문이 완료되었습니다.\n` +
+        `상품명: ${product.name}\n` +
+        `구매 수량: ${totalQuantity}\n` +
+        `발신자 이름: ${data.senderName}\n` +
+        `메시지: ${data.textMessage}`
+    );
+
+    navigate(ROUTES.HOME);
+  } catch (err) {
+    if (hasAxiosErrorStatus(err, 401)) {
+      toast.error(ERROR_MESSAGES.LOGIN_REQUIRED);
+      navigate(ROUTES.LOGIN, {
+        state: {
+          from: {
+            pathname: location.pathname,
+            search: location.search,
```

Comment:
```suggestion
            search: window.location.search,
```

브라우저 환경이므로 가급적 window를 붙여주는 것이 좋습니다!
안 붙여도 아마 전역객체가 window라 돌아가긴 할텐데요.. JS가 꼭 브라우저 환경에서만 실행되는 것은 아니니 명확하게 붙여주는 것이 좋습니다.

혹시 모를 로컬 변수로 덮이는 상황도 방지할 수 있고요. 😌

Reply:
전혀 신경쓰지 못한 부분이었습니다! 앞으로도 신경쓰며 반영하겠습니다!